File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.java
class Solution {
    public int minimumArea(int[][] grid) {
        int n=grid.length;
        int m=grid[0].length;
        int l=m,r=0,t=n,b=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(grid[i][j]==1){
                    l=Math.min(l,j);
                    r=Math.max(r,j);
                    t=Math.min(t,i);
                    b=Math.max(b,i);
                }
            }
        }
        return (r-l+1)*(b-t+1);
    }
}

File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) {
        int n=grid.size(),m=grid[0].size();
        int t=n,b=-1,l=m,r=-1;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(grid[i][j]==1){
                t=min(t,i);
                b=max(b,i);
                l=min(l,j);
                r=max(r,j);
                }
            }
        }
        return (b-t+1)*(r-l+1);
    }
};